doc/en/user/source/rest/urlchecks.rst

```diff
@@ -0,0 +1,156 @@
+
+*Response*
+
+.. code-block:: json
+
+   {"urlchecks":{"urlcheck":[
+        {"name":"external","href":"http:\/\/localhost:8080\/geoserver\/rest\/urlchecks\/external.json"},
+        {"name":"icons","href":"http:\/\/localhost:8080\/geoserver\/rest\/urlchecks\/icons.json"},
+        {"name":"safeWFS","href":"http:\/\/localhost:8080\/geoserver\/rest\/urlchecks\/safeWFS.json"}]}}
+
+
+**List all URL Checks, in XML format:**
+
+*Request*
+
+.. admonition:: curl
+
+   ::
+
+     curl -u admin:geoserver -XGET http://localhost:8080/geoserver/rest/urlchecks.xml
+
+*Response*
+
+.. code-block:: xml
+
+        <urlChecks>
+            <urlCheck>
+                <name>external</name>
+                <atom:link xmlns:atom="http://www.w3.org/2005/Atom" rel="alternate" href="http://localhost:8080/geoserver/rest/urlchecks/external.xml" type="application/atom+xml"/>
+            </urlCheck>
+            <urlCheck>
+                <name>icons</name>
+                <atom:link xmlns:atom="http://www.w3.org/2005/Atom" rel="alternate" href="http://localhost:8080/geoserver/rest/urlchecks/icons.xml" type="application/atom+xml"/>
+            </urlCheck>
+            <urlCheck>
+                <name>safeWFS</name>
+                <atom:link xmlns:atom="http://www.w3.org/2005/Atom" rel="alternate" href="http://localhost:8080/geoserver/rest/urlchecks/safeWFS.xml" type="application/atom+xml"/>
+            </urlCheck>
+        </urlChecks>
+
+
+Listing URL Check details
+-------------------------
+
+**Retrieve information about a specific URL Check:**
+
+*Request*
+
+.. admonition:: curl
+
+   ::
+
+     curl -u admin:geoserver -XGET http://localhost:8080/geoserver/rest/urlchecks/icons.xml
+
+*Response*
+
+.. code-block:: xml
+
+        <urlCheck>
+            <name>icons</name>
+            <description>External graphic icons</description>
+            <enabled>true</enabled>
+            <regex>^https://styles.server.net/icons/.*$</regex>
+        </urlCheck>
+
+
+Creating a URL Check
+--------------------
+
+**Create a new URL Check:**
+
+*Request*
+
+.. admonition:: curl
+
+   ::
+
+     curl -u admin:geoserver -XPOST -H "Content-type: text/xml" \
+     -d "<urlCheck> \
+            <name>icons</name> \
+            <description>External graphic icons</description> \
+            <enabled>true</enabled> \
+            <regex>^https://styles\.server\.net/icons/.*$</regex> \
+        </urlCheck>" \
+     http://localhost:8080/geoserver/rest/urlchecks
+
+*Response*
+
+::
+
+   201 Created
+
+Changing an existing URL Check
+------------------------------
+
+**Edit the configuration of an existing URL Check:**
+
+*Request*
+
+.. admonition:: curl
+
+   ::
+
+     curl -u admin:geoserver -XPUT -H "Content-type: text/xml" \
+     -d "<urlCheck> \
+            <description>External graphic icons (disabled) </description> \
+            <enabled>false</enabled> \
+            <regex>^https://styles\.server\.com/icons/.*$</regex> \
+        </urlCheck>" \
```

Comment:
Thinking out loud, what happens when a different type of URl check is added to the API (e.g., @jodygarnett wanted to also have simpler checks avoiding regexes, in the future, like prefix validators for example). If the container item is "urlCheck" I believe we have a problem, how is the parser going to tell apart a regex validator from another one?

Reply:
Revised to be specific to the only url check implemented at the moment.
Now the parser could be instructed to identify different types of checks.